java/pom.xml

```diff
@@ -81,8 +81,8 @@ under the License.
   </scm>
 
   <issueManagement>
-    <system>Jira</system>
-    <url>https://issues.apache.org/jira/browse/arrow</url>
+    <system>GitHub</system>
```

Comment:
Because `maven/pom.xml` and `bom/pom.xml` do not inherit from the top-level `pom.xml`. you may want to change those as well (note that `maven/pom.xml` is being removed as part of #43139)

Reply:
Great catch, I didn't think to look for other uses.

Reply:
Turns out we don't populate `<issueManagement>` in the other poms. Is it typical to add it?

Reply:
I am actually not sure whether we need this? 
cc @lidavidm

Reply:
It's clearly not strictly necessary, so I'd rather we just take the update here?

Reply:
Not necessary but nice to have. It may be used by tools and shows up in the site generated by Maven or also accessible from websites like Maven Central.

You can check and compare the pages for [arrow-memory-core](https://central.sonatype.com/artifact/org.apache.arrow/arrow-memory-core) and [arrow-bom](https://central.sonatype.com/artifact/org.apache.arrow/arrow-bom)

Reply:
Looks like `Project URL` and `Source Control` are also not configured correctly for maven central. Thanks for sharing.